x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -1165,9 +1162,33 @@ static long machineMemoryFromStats(OsStats stats) {
     @Override
     public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
         return List.of(
-            new SystemIndexDescriptor(MlMetaIndex.indexName(), "Contains scheduling and anomaly tracking metadata"),
-            new SystemIndexDescriptor(MlConfigIndex.indexName(), "Contains ML configuration data"),
-            new SystemIndexDescriptor(InferenceIndexConstants.INDEX_PATTERN, "Contains ML model configuration and statistics")
+            SystemIndexDescriptor.builder()
+                .setIndexPattern(MlMetaIndex.indexName() + "*")
+                .setPrimaryIndex(MlMetaIndex.indexName())
+                .setDescription("Contains scheduling and anomaly tracking metadata")
+                .setMappings(MlMetaIndex.mapping())
+                .setSettings(MlMetaIndex.settings())
+                .setVersionMetaKey("version")
+                .setOrigin(ML_ORIGIN)
+                .build(),
+            SystemIndexDescriptor.builder()
+                .setIndexPattern(MlConfigIndex.indexName() + "*")
+                .setPrimaryIndex(MlConfigIndex.indexName())
+                .setDescription("Contains ML configuration data")
+                .setMappings(MlConfigIndex.mapping())
+                .setSettings(MlConfigIndex.settings())
+                .setVersionMetaKey("version")
+                .setOrigin(ML_ORIGIN)
+                .build(),
+            SystemIndexDescriptor.builder()
+                .setIndexPattern(InferenceIndexConstants.INDEX_PATTERN)
+                .setPrimaryIndex(InferenceIndexConstants.LATEST_INDEX_NAME)
+                .setDescription("Contains ML model configuration and statistics")
+                .setMappings(InferenceIndexConstants.mapping())
+                .setSettings(InferenceIndexConstants.settings())
+                .setVersionMetaKey("version")
+                .setOrigin(ML_ORIGIN)
+                .build()
```

Comment:
Inference does ever update the existing mappings of the old index. So, on a new node where `InferenceIndexConstants.LATEST_INDEX_NAME` is newer than other nodes in the index, a NEW index is created. 

Do system indices still apply the latest index mapping, even if that new node is not master? Previously we got around this by manually adding the template in teh analytics job assignment, but this PR removes that.

Reply:
You are right, it won't work as I have it at the moment.  I think the solution will be to issue a create index request with the mappings from the upgraded node and origin set to `ML_ORIGIN` where previously we created the template.  That will be permitted by the master node even though the mappings don't match the old master's outdated view of what the mappings will be.  I will update the code to do that.